internal/service/s3control/account_public_access_block_data_source_test.go

```diff
@@ -0,0 +1,119 @@
+package s3control_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3control"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func TestAccS3ControlAccountPublicAccessBlockDataSource_basic(t *testing.T) {
+	var conf s3control.PublicAccessBlockConfiguration
+	resourceName := "aws_s3_account_public_access_block.test"
+	dataSourceName := "data.aws_s3_account_public_access_block.test"
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acctest.PreCheck(t) },
+		ErrorCheck:        acctest.ErrorCheck(t, s3control.EndpointsID),
+		ProviderFactories: acctest.ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccS3ControlAccountPublicAccessBlockDataSource_basic(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckAccountPublicAccessBlockExistsDataSource(resourceName, &conf),
```

Comment:
Don't need to do the validation for the resource here

```suggestion

Reply:
removed the validation